modules/collect/src/main/java/com/opengamma/collect/timeseries/LocalDateDoubleTimeSeries.java

```diff
@@ -1,293 +1,106 @@
-    ArgChecker.noNulls(dates, "dates");
-    ArgChecker.notNull(values, "values");
-    validate(dates, values);
-    this.dates = dates.clone();
-    this.values = values.clone();
-  }
-
-  /**
-   * Creates an instance without validating the supplied arrays.
+   * Returns a map of times and values in this time-series.
    *
-   * @param dates  the dates
-   * @param values  the values
-   * @param trusted  flag to distinguish constructor
-   */
-  private LocalDateDoubleTimeSeries(LocalDate[] dates, double[] values, boolean trusted) {
-    // constructor exists to avoid clones where possible
-    // because Joda-Beans owns the main constructor, this one has a weird flag
-    // use createUnsafe() instead of calling this directly
-    this.dates = dates;
-    this.values = values;
-  }
-
-  /**
-   * Creates an instance validating the supplied arguments.
-   *
-   * @param date  the date
-   * @param value  the value
-   */
-  private LocalDateDoubleTimeSeries(LocalDate date, double value) {
-    dates = new LocalDate[] {ArgChecker.notNull(date, "date")};
-    values = new double[] {value};
-  }
-
-  //-----------------------------------------------------------------------
-  /**
-   * Gets the dates in the series.
-   * The dates are ordered from earliest to latest.
-   * @return the value of the property, not null
-   */
-  private LocalDate[] getDates() {
-    return dates.clone();
-  }
-
-  /**
-   * Gets the values in the series.
-   * The date for each value is at the matching array index.
-   * @return the value of the property, not null
+   * @return a map of the elements of this time-series
    */
-  private double[] getValues() {
-    return values.clone();
-  }
+  public abstract ImmutableSortedMap<LocalDate, Double> toMap();
 
   /**
    * Returns the list of dates in this time-series.
    * <p>
    * This provides low-level access to the internal data.
-   * The stream and lambda methods should be used in preference to this method.
+   * The stream and lambda methods should be used in preference
+   * to this method. There is no guarantee that the underlying
+   * implementation holds the dates in a list.
    *
    * @return the list of dates in this time-series
    */
-  public ImmutableList<LocalDate> dates() {
-    return ImmutableList.copyOf(dates);
-  }
+  public abstract ImmutableList<LocalDate> dates();
 
   /**
    * Returns the list of values in this time-series.
    * <p>
    * This provides low-level access to the internal data.
-   * The stream and lambda methods should be used in preference to this method.
+   * The stream and lambda methods should be used in preference
+   * to this method. There is no guarantee that the underlying
+   * implementation holds the values in a list.
    *
    * @return the list of values in this time-series
    */
-  public ImmutableList<Double> values() {
-    return ImmutableList.copyOf(Doubles.asList(values));
-  }
+  public abstract ImmutableList<Double> values();
```

Comment:
Rename dateStream() and valueStream() to dates() and values().

If list versions are needed, they should be named dateList() and valueList() but I suspect they are not needed